src/OSPSuite.Core/Domain/CreationMetaData.cs

```diff
@@ -29,14 +29,20 @@ public enum CreationMode
    public class CreationMetaData
    {
       private DateTime _createdAt;
-      public virtual string CreatedBy { get; private set; }
+      public virtual string CreatedBy { get; }
 
       public virtual Origin Origin { get; set; }
-
+      
       /// <summary>
       /// Version of the application (typically X.Y.Z)
       /// </summary>
       public virtual string Version { get; set; }
+
+      /// <summary>
+      /// Internal version that can be referenced to check the real internal version of the project
+      /// </summary>
+      public virtual int? InternalVersion { get; set; }
```

Comment:
New field int? for CreationMetaData (even if we do not use it quite yet, this could be quite useful for the future)